src/main/java/pages/IndexPage.java

```diff
@@ -0,0 +1,159 @@
+    }
+
+    public LeftSideMenu getLeftSideMenu() {
+        return leftSideMenu;
+    }
+
+    //Login section
+    @FindBy(css = "li.dropdown.uui-profile-menu")
+    private WebElement userIcon;
+
+    @FindBy(css = "[id = 'name']")
+    private WebElement userName;
+
+    @FindBy(css = "[id = 'password']")
+    private WebElement userPassword;
+
+    @FindBy(css = "[id = 'login-button']")
+    private WebElement loginButton;
+
+    @FindBy(css = "div.logout")
+    private WebElement logoutButton;
+
+    @FindBy (id = "user-name")
+    private WebElement fullUserName;
+
+    //Icon section
+    @FindBy (css = ".icons-benefit.icon-practise")
+    private WebElement iconPractice;
+
+    @FindBy (css = ".icons-benefit.icon-custom")
+    private WebElement iconCustom;
+
+    @FindBy (css = ".icons-benefit.icon-multi")
+    private WebElement iconMulti;
+
+    @FindBy (css = ".icons-benefit.icon-base")
+    private WebElement iconBase;
+
+    @FindBy (css = ".benefit-txt")
+    private List<WebElement> textsUnderIcons;
+
+    //iFrame
+    @FindBy (id = "frame")
+    private WebElement iframe;
+
+    @FindBy (css = "#frame-button")
+    private WebElement iframeButton;
+
+    public void openIndexPage() {
+        driver.navigate().to("https://jdi-testing.github.io/jdi-light/index.html");
+    }
+
+    public IndexPage openIndexPageFluent() {
+        driver.navigate().to("https://jdi-testing.github.io/jdi-light/index.html");
+        return this;
+    }
+
+    public void login(String name, String password) {
+        userIcon.click();
+        userName.sendKeys(name);
+        userPassword.sendKeys(password);
+        loginButton.click();
+        Assert.assertTrue(logoutButton.isDisplayed(), "Logout button isn't displayed");
+    }
+
+    public IndexPage loginFluent(String name, String password) {
+        userIcon.click();
+        userName.sendKeys(name);
+        userPassword.sendKeys(password);
+        loginButton.click();
+        Assert.assertTrue(logoutButton.isDisplayed(), "Logout button isn't displayed");
+        return this;
+    }
+
+
+
+    public void checkPageIconsAndText() {
+        //Icons check
+        Assert.assertTrue(iconPractice.isDisplayed());
+        Assert.assertTrue(iconCustom.isDisplayed());
+        Assert.assertTrue(iconMulti.isDisplayed());
+        Assert.assertTrue(iconBase.isDisplayed());
+
+        //Text check
+        assertEquals(textsUnderIcons.get(0).getText(), "To include good practices\n"
+                + "and ideas from successful\n"
+                + "EPAM project");
+        assertEquals(textsUnderIcons.get(1).getText(), "To be flexible and\n"
+                + "customizable");
+        assertEquals(textsUnderIcons.get(2).getText(), "To be multiplatform");
+        assertEquals(textsUnderIcons.get(3).getText(), "Already have good base\n"
+                + "(about 20 internal and\n"
+                + "some external projects),\n"
+                + "wish to get more…");
+    }
```

Comment:
These texts should not be hardcoded in Page Object methods. Better to use data providers or to store this data in class variables or in some other test data storages

Reply:
moved to DataElementNames class